### PR TITLE
[PWA-1204] Inlining page data and webpack chunks

### PR DESCRIPTION
**Pull request by justinconabree**
## Description
- Adding ComputedInterface
- Adding to pwa di.xml to include two new types to the Computed pool
- Adding PageInfo model to resolve page information based on request
- Adding Computed type: PageType which exports a json of page information
- Adding Computed type: WebpackChunks which finds root component JS files based on page type
- Handling additional page information requested by upward config based on page type

Related stories:
- https://jira.corp.magento.com/browse/PWA-1906
- https://jira.corp.magento.com/browse/PWA-1204

Previous PR comments applied: https://github.com/magento/magento2-upward-connector/pull/34

### CODE REVIEW

File: Model/PageType.php [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                throw new \RuntimeException(
+                    __('No such entity found with matching URL key: %url', ['url' => $url])
+                );
+            }
+
+            $result = $resultArray ?: null;
+        }
+
+        return $result;
+    }
```
Comment (larsroettig) [inline Model/PageType.php:156]: Pls cover this with a test

**Review by larsroettig — CHANGES_REQUESTED** [review]

File: Model/Computed/PageInfoNonce.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+     * @return string
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function generateNonce(DefinitionIterator $iterator): string
+    {
+        $pageInfo = $this->pageTypeResolver
+            ->setContext($iterator->getContext())
+            ->getInfo();
+
+        $nonceData = uniqid('', true);
```
Comment (AlexMaxHorkun) [inline Model/Computed/PageInfoNonce.php]: uniqid() does not generate cryptographically secure random value. Use something like random_bytes() or Magento's Math\Random helper to generate random string.

File: Model/Computed/PageInfoNonce.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        if ($pageInfo) {
+            $nonceData = $pageInfo['type'] . $pageInfo['id'] . $nonceData;
+        }
+
+        return base64_encode(
+            $this->encryptor->encrypt($nonceData)
```
Comment (AlexMaxHorkun) [inline Model/Computed/PageInfoNonce.php]: why waste processor time encrypting a random value? There's nothing secret about it. Just base64 encode it. Or better yet use Random::randomString() result directly with a lengthy enough value

**Review by AlexMaxHorkun — CHANGES_REQUESTED** [review]

File: Model/Computed/PageInfoNonce.php (hunk 2 above)
Comment (justinconabree) [reply Model/Computed/PageInfoNonce.php]: Fixed! Changed to Math\Random getRandomString()

File: Model/Computed/PageInfoNonce.php (hunk 3 above)
Comment (justinconabree) [reply Model/Computed/PageInfoNonce.php]: I had done this to add the encryption key to the randomness, but with your comment I've simplified it to just return the Math\Random string with a length of 45 char (minimum for CSP is 32, so we should be fine)

**Comment by tjwiebell** [comment]
@magento import pr to magento-commerce/magento2-upward-connector

**Comment by tjwiebell** [comment]
@magento import pr to magento-commerce/magento2-upward-connector

**Comment by tjwiebell** [comment]
@magento import pr to magento-commerce/magento2-upward-connector

**Pull request closed without merging**